src/python/WMCore/MicroService/MSUnmerged/MSUnmerged.py

```diff
@@ -57,27 +70,144 @@ def execute(self):
             self.logger.exception(msg)
             self.updateReportDict(summary, "error", msg)
 
-        # this one is put here just for example.
-        self.updateReportDict(summary, "error", 42)
+        try:
+            totalNumRses, totalNumFiles, numRsesCleaned, numFilesDeleted = self._execute(rseList)
+            msg = "\nTotal number of processed RSEs: %s."
+            msg += "\nTotal number of files to be deleted: %s."
+            msg += "\nNumber of RSEs cleaned: %s."
+            msg += "\nNumber of files deleted: %s."
+            self.logger.info(msg,
+                             totalNumRses,
+                             totalNumFiles,
+                             numRsesCleaned,
+                             numFilesDeleted)
+            self.updateReportDict(summary, "total_num_rses", totalNumRses)
+            self.updateReportDict(summary, "total_num_files", totalNumFiles)
+            self.updateReportDict(summary, "num_rses_cleaned", numRsesCleaned)
+            self.updateReportDict(summary, "num_files_deleted", numFilesDeleted)
+        except Exception as ex:
+            msg = "Unknown exception while running MSUnmerged thread Error: %s"
+            self.logger.exception(msg, str(ex))
+            self.updateReportDict(summary, "error", msg)
+
         return summary
 
+    def _execute(self, rseList):
+        """
+        Executes the MSUnmerged pipelines
+        :param rseList: A list of RSEs to work on
+        :return:        a tuple with:
+                            total number of RSEs
+                            total number of files found for deletion
+                            number of RSEs cleaned
+                            number of files deleted
+        """
+        totalNumRses = 0
+        totalNumFiles = 0
+        numRsesCleaned = 0
+        numFilesDeleted = 0
+
+        # Call the workflow dispatcher:
+        for rse in rseList:
+            try:
+                rse = MSUnmergedRSE(rse)
+                self.plineUnmerged.run(rse)
+                msg = "\n----------------------------------------------------------"
+                msg += "\nMSUnmergedRSE: %s"
+                msg += "\n----------------------------------------------------------"
+                self.logger.debug(msg, pformat(rse))
+                totalNumRses += 1
+                if self._checkClean(rse):
+                    numRsesCleaned += 1
+                totalNumFiles += rse['counters']['totalNumFiles']
+                numFilesDeleted += rse['counters']['numFilesDeleted']
+            except Exception as ex:
+                msg = "%s: General error from pipeline. RSE: %s. Error:  \n%s. "
+                msg += "\nWill retry again in the next cycle."
+                self.logger.exception(msg, self.plineUnmerged.name, rse['rse'], str(ex))
+                continue
+        return totalNumRses, totalNumFiles, numRsesCleaned, numFilesDeleted
+
+    def filterRseList(self, rseList):
```

Comment:
Doesn't the method `getRSEList` does all the job already? If not, how are you considering to use this one?

Reply:
I was thinking about applying the the rest of the load constraints constraints here, those like `self.msConfig.limitRSEsPerInstance` etc. We can skip it if we want, or preserve  the placeholder for whatever constrain comes next.

Reply:
I think I would rely on the expression only. If a POD is defined to work in a set of RSEs, we better make sure that very same set is cleaned up by the same POD.
Regarding placeholders for a _possible_ development, you know what my opinion is on that. Just ditch it, please ;)

Reply:
removing it :)

Reply:
done